apps/chat/src/store/files/files.reducers.ts

```diff
@@ -259,21 +259,31 @@ export const filesSlice = createSlice({
       }>,
     ) => {
       state.newAddedFolderId = undefined;
+
+      const targetFolder = state.folders.find((f) => f.id === payload.folderId);
+
+      if (!targetFolder) return;
+
       state.folders = state.folders.map((folder) => {
-        if (folder.id !== payload.folderId) {
-          return folder;
-        }
+        if (folder.id === payload.folderId) {
+          return {
+            ...folder,
+            name: payload.newName.trim(),
+            id: constructPath(targetFolder.folderId, payload.newName),
+          };
+        } else if (folder.id.startsWith(payload.folderId)) {
```

Comment:
maybe we should check 
```suggestion
        } else if (folder.id.startsWith(`${payload.folderId}/`)) {
```
to prevent situation if folder has paths like: folder1/folder12/  and folder1/folder1/ and payload.folderId === folder1/folder1

Reply:
good catch. thanks. done.